src/pal/src/file/file.cpp

```diff
@@ -3605,18 +3721,26 @@ GetTempFileNameW(
             goto done;
         }
     }
-       
+    
+    tempfile_name = (CHAR *)alloca((MAX_LONGPATH+1) * sizeof(CHAR));
```

Comment:
Here, we are allocating MAX_LONGPATH size, so there's no runtime dependency on what variant size to allocate. So putting it on heap.